endpoints/openrtb2/auction.go

```diff
@@ -1378,42 +1394,44 @@ func (deps *endpointDeps) processStoredRequests(ctx context.Context, requestJson
 	// Apply any Stored Imps, if they exist. Since the JSON Merge Patch overrides arrays,
 	// and Prebid Server defers to the HTTP Request to resolve conflicts, it's safe to
 	// assume that the request.imp data did not change when applying the Stored BidRequest.
-	impExtInfoMap := make(map[string]exchange.ImpExtInfo, len(impIds))
-	for i := 0; i < len(impIds); i++ {
-		resolvedImp, err := jsonpatch.MergePatch(storedImps[impIds[i]], imps[idIndices[i]])
+	impExtInfoMap := make(map[string]exchange.ImpExtInfo, 0)
+	resolvedImps := make([]json.RawMessage, 0)
+	for i, impData := range impInfo {
+		if impData.ImpExtPrebid.StoredRequest != nil && len(impData.ImpExtPrebid.StoredRequest.ID) > 0 {
+			resolvedImp, err := jsonpatch.MergePatch(storedImps[impData.ImpExtPrebid.StoredRequest.ID], impData.Imp)
 
-		if err != nil {
-			hasErr, Err := getJsonSyntaxError(imps[idIndices[i]])
-			if hasErr {
-				err = fmt.Errorf("Invalid JSON in Imp[%d] of Incoming Request: %s", i, Err)
-			} else {
-				hasErr, Err = getJsonSyntaxError(storedImps[impIds[i]])
+			if err != nil {
+				hasErr, Err := getJsonSyntaxError(impData.Imp)
```

Comment:
I think this is a good time to fix the improper use of `Err` as a sneaky way of avoiding shadowing with the `err` variable.

Reply:
`Err` is a string description of error if it's present which I believe should be a nicer representation of `err.Error()`. 
It's handled further in code like this:
```
hasErr, Err := getJsonSyntaxError(impData.Imp)
				if hasErr {
					err = fmt.Errorf("Invalid JSON in Imp[%d] of Incoming Request: %s", i, Err)
...
```
Should it be modified to something like this:
```
err = fmt.Errorf("Invalid JSON in Imp[%d] of Incoming Request: %s, original error: %s", i, Err, err.Error())
```

Reply:
Renamed to `errMessage`